lib/translator/inquiry.cc

```diff
@@ -236,6 +232,70 @@ scsi_defs::UnitSerialNumber BuildUnitSerialNumberVpd() {
   return TranslateUnitSerialNumberVpdResponse(identify_namespace_data);
 }
 
+// Builds the IdentificationDescriptor according to 6.1.4.5 of the translation
+// spec: https://nvmexpress.org/wp-content/uploads/NVM_Express_-_SCSI_Translation_Reference-1_5_20150624_Gold.pdf
+// This only builds the constant part of the Identification Descriptor. The 
+// caller is responsible for building the identifier.
+scsi_defs::IdentificationDescriptor BuildIdentificationDescriptor(
+  nvme_defs::IdentifyNamespace &identify_namespace_data) {
+  scsi_defs::IdentificationDescriptor identification_descriptor = 
+    scsi_defs::IdentificationDescriptor();
+  identification_descriptor.protocol_identifier = 0x0;
+  identification_descriptor.code_set = 0x1;
+  identification_descriptor.piv = 0;
+  identification_descriptor.association = 0b00;
+  identification_descriptor.identifier_type = 0x2;
+
+  // TODO:basimsahaf return the identifier at the end of the buffer. To be done 
+  // after we figure out how to send data back to SCSI. EUI-64 field is a 
+  // variable field hence its left out for now
+  
+  // check if nonzero
+  bool nguid_nz =
+      identify_namespace_data.nguid[0] || identify_namespace_data.nguid[1];
+  bool eui64_nz = identify_namespace_data.eui64;
+
+  if (nguid_nz) {
+    identification_descriptor.identifier_length = 0x10;
+    uint8_t identification_descriptor_identifier[16];
+    memcpy(&identification_descriptor_identifier, &identify_namespace_data.nguid,
+      16);
+  } else if(eui64_nz) {
+    identification_descriptor.identifier_length = 0x8;
```

Comment:
make constant for len

Reply:
or sizeof ui64

Reply:
looks like I can't use `sizeof` for a bitfield. Do you suggest making an enum for it or a constant?

Reply:
If the constant is just for inquiry.cc you could put it in the anonymous namespace so it's not exposed to the rest of the library

Reply:
sounds good! I put the constant inside the anon namespace